main.go

```diff
@@ -116,12 +126,25 @@ func main() {
 		}()
 	}
 
-	server.StatsCollect()
+	stats.Collect()
 	if err := s.Run(); err != nil {
 		log.Fatalf("skydns: %s", err)
 	}
 }
 
+func loadConfig(client *etcd.Client, config *server.Config) error {
```

Comment:
I tried it but it required gymnastics on initializing the getter (to set priority and ttl once when the getter was created, and then again after the config was loaded).  It felt less coupled this way (I briefly created a second getter but it didn't feel special enough to be independent).

The etcd logic in main could probably move into backends/etcd as helper code though.

> On Mar 8, 2015, at 10:34 AM, Miek Gieben <notifications@github.com> wrote:
> 
> In main.go:
> 
> >  	if err := s.Run(); err != nil {
> >  		log.Fatalf("skydns: %s", err)
> >  	}
> >  }
> >  
> > +func loadConfig(client *etcd.Client, config *server.Config) error {
> I wonder if we should make loadconfig part of the Getter interface?
> 
> —
> Reply to this email directly or view it on GitHub.
>